### PR TITLE
Cascaders Online

### LINKED ISSUE #15: Cascaders Online
@ggilligan12 and I were discussing Cascaders. He came up with the idea of adding it to mapp.

Plan of action:
- a button to make yourself appear on the map for everyone
    - a checkbox to display or hide your name
- a freeform text box to describe what exactly you are happy to help with
- the button should be toggleable, so you can temporarily disable "cascader mode" without losing the freeform text (basically a DnD for cascading, as you may still want your friends to see you)
- a way to ban users who abuse the freeform text box

### PR DESCRIPTION
**Pull request by qaisjp**
This fixes #15 and solves 3 of the 4 requirements.

The hide-name-for-everyone feature will come at a later time, after we see wholesome use of the tagline feature.

Demo available at https://mapp-dev.betterinformatics.com.

### CODE REVIEW

**Comment by qaisjp** [comment]
there should be a max length for the tagline.

`len("DICE, Python, Artificial Intelligence, Everything, Machine Learning, and definitely not Mathematics!")` should be long enough. that's 100 characters.

File: map/views.py [hunk 1]
```diff
@@ -30,6 +32,38 @@ def handle_invalid_usage(error):
     return response
 
 
+def get_cascaders() -> List[str]:
+    return list(flask_redis.smembers("cascaders.users"))
+
+
+def find_cascader(cascaders: List[str], hash: str) -> Optional[str]:
```
Comment (neanias) [inline map/views.py]: I know you’re not using it here, but it’s good practice not to override builtin methods (like `hash`)

File: map/views.py [hunk 2]
```diff
@@ -30,6 +32,38 @@ def handle_invalid_usage(error):
...
+    for uun in cascaders:
+        if check_uun_hash(uun, hash):
+            return uun
+    return None
+
+
+def get_cascader_elsewhere_count(cascaders: List[str], notRoom: str) -> int:
+    if not current_user.is_authenticated:
+        return 7
```
Comment (neanias) [inline map/views.py]: Why 7?

Comment (qaisjp) [reply map/views.py]: Random number, for demo

Comment (neanias) [reply map/views.py]: Hmm, ok. Better to not have too many magic numbers in the source code. Maybe flag this up to the user somehow?

**Comment by neanias** [comment]
What's this Travis CI stuff about?

**Comment by qaisjp** [comment]
Not sure, I think it's a side effect from #14

File: map/views.py (hunk 2 above)
Comment (qaisjp) [reply map/views.py]: Fixed — this didn't work anyway.

**Comment by neanias** [comment]
My bad, sorry. I'll see where the restrictions are

**Review by neanias — APPROVED** [review]

**Pull request merged**